pkgs/development/compilers/bambu/default.nix

```diff
@@ -0,0 +1,257 @@
+  (gcc:
+  lib.assertMsg (gcc.system == "i686-linux") "Bambu frontends have to be i686 but ${gcc.name} is ${gcc.system}")
+  gccFrontends;
+# Assert that the clang and GCC frontends versions are compatible with bambu
+assert builtins.all
+  (llvmPackages:
+  lib.assertMsg (builtins.elem (lib.versions.major llvmPackages.clang.version) supportedLlvmMajorVersions) "Clang version ${llvmPackages.clang.version} is not supported as a frontend by bambu. The following major clang versions are supported: ${builtins.concatStringsSep " " supportedLlvmMajorVersions}")
+  clangFrontends;
+assert builtins.all
+  (gcc:
+  lib.assertMsg (builtins.elem (lib.versions.major gcc.version) supportedGccMajorVersions) "GCC version ${gcc.version} is not supported as a frontend by bambu. The following major GCC versions are supported: ${builtins.concatStringsSep " " supportedGccMajorVersions}")
+  gccFrontends;
+# Assert that there is only one compiler frontend for each major version of the compiler
+let
+  passedClangVersions = (builtins.map (llvmPackages: llvmPackages.clang.version) clangFrontends);
+  duplicateClangVersion = lib.lists.findFirst (clangVersion: (builtins.length (builtins.filter (otherClangVersion: (lib.versions.major clangVersion) == (lib.versions.major otherClangVersion)) passedClangVersions)) > 1) false passedClangVersions;
+  passedGccVersions = (builtins.map (gcc: gcc.version) gccFrontends);
+  duplicateGccVersion = lib.lists.findFirst (gccVersion: (builtins.length (builtins.filter (otherGccVersion: (lib.versions.major gccVersion) == (lib.versions.major otherGccVersion)) passedGccVersions)) > 1) false passedGccVersions;
+in
+assert lib.assertMsg (duplicateClangVersion == false) "Multiple clang frontends with the same major version (${lib.versions.major duplicateClangVersion}) are not supported by bambu. You passed packages with the following versions: ${builtins.concatStringsSep " " (builtins.filter (clangVersion: (lib.versions.major clangVersion) == (lib.versions.major duplicateClangVersion)) passedClangVersions)}";
+assert lib.assertMsg (duplicateGccVersion == false) "Multiple gcc frontends with the same major version (${lib.versions.major duplicateGccVersion}) are not supported by bambu. You passed packages with the following versions: ${builtins.concatStringsSep " " (builtins.filter (gccVersion: (lib.versions.major gccVersion) == (lib.versions.major duplicateGccVersion)) passedGccVersions)}";
+stdenv.mkDerivation
+{
+  pname = "bambu";
+  version = "unstable-2023-08-01";
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub {
+    owner = "ferrandi";
+    repo = "PandA-bambu";
+    rev = "1d24ce0f8fe86686dc2366062a8d9707b685b3e6";
+    hash = "sha256-yIe2sRxbfVNvK39Y9zPLKaHwnxiI58jhJkVN/j96Q0A=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    autoconf-archive
+    automake
+
+    libtool
+
+    bison
+    doxygen
+    flex
+    graphviz
+    verilog
+    verilator
+  ];
+
+
+  buildInputs = [
+    boost
+    boost.dev
```

Comment:
Are all these `.dev` necessary?

Reply:
As far as I remember yes, but I am not 100% sure. I will try if it works without them.

Reply:
buildInputs already chooses `.dev` so there are not necessary.